futures-util/src/stream/stream/buffered.rs

```diff
@@ -40,6 +40,16 @@ where
     }
 }
 
+impl<St> FusedStream for Buffered<St>
+where
+    St: FusedStream,
+    St::Item: Future,
+{
+    fn is_terminated(&self) -> bool {
+        self.stream.is_terminated() && self.in_progress_queue.is_empty()
```

Comment:
I don't think this is quite right-- the stream might have terminated previously, and then the queue yielded a final element, but has not yet yielded `None`. In this case, the `Stream` still has to yield `None`, but this will return `true`

Reply:
Why does a stream have to yield a `None` before terminating?

Reply:
it's part of the contract of `FusedStream` that `Stream`s which are not "stalled" should return `None` before being considered "terminated", similar to how `FusedIterator`s must return `None`.

Reply:
Kk i'll fix that.

But what is the rational behind this contract? Why would you want to pretend that a stream is not yet terminated until its polled with None?

Reply:
The primary purpose of `FusedStream` is for use in `select!` to prevent polling a stream after it has yielded `None`, which violates the contract of the `Stream`. Users of `select!` would still generally expect to see a `None` value appear at some point and be able to handle the event of the stream becoming terminated, rather than it just stopping without ever yielding `None`.

Reply:
Forgot about that PR, should be fixed I think.